src/iexamine.cpp

```diff
@@ -3497,6 +3498,80 @@ void iexamine::climb_down( player &p, const tripoint &examp )
     g->m.creature_on_trap( p );
 }
 
+void iexamine::autodoc( player &p, const tripoint &examp ) {
+    enum options {
+        CONFIGURE,
+        INSTALL_CBM,
+        CANCEL,
+    };
+
+    std::string status = uistate.iexamine_autodoc_config ? _( "Online" ) :
+                         _( "Configuration required" );
+
+    uimenu amenu;
+    amenu.selected = 0;
+    amenu.text = string_format( _( "Autodoc Mk. XI.  Status: %s.  Please choose operation." ), status );
+    amenu.addentry( CONFIGURE, true, 'c', str_to_illiterate_str( _( "Configure Autodoc." ) ) );
+    amenu.addentry( INSTALL_CBM, uistate.iexamine_autodoc_config, 'i',
+                    str_to_illiterate_str( _( "Choose Compact Bionic Module to install." ) ) );
+
+    amenu.addentry( CANCEL, true, 'q', str_to_illiterate_str( _( "Do nothing." ) ) );
+
+    amenu.query();
+
+    bool adjacent_couch = false;
+    bool in_position = false;
+    for( const auto &couch_loc : g->m.points_in_radius( examp, 1, 0 ) ) {
+        if( g->m.furn( couch_loc ).obj().id == "f_autodoc_couch" ) {
+            adjacent_couch = true;
+            if( p.pos() == couch_loc ) {
+                in_position = true;
+            }
+        }
+    }
+
+    switch( static_cast<options>( amenu.ret ) ) {
+        case CONFIGURE:
+            uistate.iexamine_autodoc_config = true;
+            //Flavor message to justify using player skills and INT below in install_bionics for determining success chance
+            p.add_msg_if_player( m_good,
+                                 _( "Using your knowledge of human anatomy, electronics and mechanics, you enter various data into the Autodoc console, configuring it to make bionic manipulations." ) );
+            p.mod_moves( -300 );
+            return;
+
+        case INSTALL_CBM: {
+            if( !adjacent_couch ) {
+                popup( _( "No connected couches found.  Operation impossible. Exiting." ) );
+                return;
+            }
+            if( !in_position ) {
+                popup( _( "No patient found located on the connected couches.  Operation impossible. Exiting." ) );
+                return;
+            }
+
+            const int bionic = g->inv_for_flag( "CBM", _( "Choose CBM to install" ) );
+            const item &it = p.i_at( bionic );
```

Comment:
The two references `it` and `itemtype` are only needed below the `if` block, create them there (right before they are needed).